nestor_api/lib/git.py

```diff
@@ -18,77 +17,74 @@ def branch(repository_dir, branch_name):
     io.execute(f'git checkout{"" if exists else " -b"} {branch_name}', repository_dir)
 
 
-def create_working_repository(app_name):
-    """Checkout a branch of a repository"""
-    # First, update the pristine repository for this app
-    pristine_directory = update_pristine_repository(app_name)
+def create_working_repository(app_name: str, git_url: str) -> str:
+    """Create a working copy of an app's repository"""
+    pristine_directory = update_pristine_repository(app_name, git_url)
 
     repository_dir = io.create_temporary_copy(pristine_directory, app_name)
 
     Logger.debug(
-        {app_name, repository_dir}, "[git#create_working_directory] Created a working repository",
+        {"app": app_name, "repository": repository_dir},
+        "[git#create_working_directory] Created a working repository",
     )
 
     return repository_dir
 
 
-def get_last_tag(repository_dir):
+def get_last_tag(repository_dir: str) -> str:
     """Retrieves the last tag of a repository (locally)"""
-    return io.execute("git describe --always --abbrev=0", repository_dir)
+    return io.execute("git describe --always --abbrev=0", repository_dir).rstrip()
```

Comment:
I am wondering if we should not put `.rstrip()` directly in `io.execute()` method.
Do you think of any case where we would want to keep trailing whitespaces?

Reply:
I don't see any 🤔 , I'll put it in `io.execute()` then.